File: ruby/styles/hidden.rb
require_relative 'base'

module Styles
  class Hidden < Base
    def index?
      false
    end

    def prepare( compiler, article, parents)
      unless parents.size == 1
        article.error( 'Hidden page must be at root of website')
      end
    end

    def post_process_html( root_url, article, html)
      html = super
      html = html.gsub( /href="[^"]*"/i) do |match|
        if /:/ =~ match
          match
        else
          match.sub('"','"' + root_url)
        end
      end
      html.gsub( 'url("./', 'url("' + root_url)
    end
  end
end

File: java/src/com/alofmethbin/jadawin/styles/Hidden.java
package com.alofmethbin.jadawin.styles;

import com.alofmethbin.jadawin.Article;

public class Hidden extends BaseStyle {

    @Override
    public boolean isIndex() {
        return false;
    }

    @Override
    public String postProcessHTML( Article article, String html) {
        html = super.postProcessHTML( article, html);
        return html.replaceAll( "", "");
    }

    @Override
    public void prepare( Article article) {
        if (article.parent() == null) {
            article.error( "Hidden page must be at root of website");
        }

        if ((article.parent() != null) && (article.parent().parent() != null)) {
            article.error( "Hidden page must be at root of website");
        }
    }
}
